### PR TITLE
 Fixes #313 :- Hashtags on home page

### LINKED ISSUE #313: Start Page: Show top hashtags from last 24 hours
Instead of "Distributed Social Media Message Search Engine" display an area with the top hashtags of the last 24 hours (in simple text style).

Show 12 hashtags here as a link.

screenshot from 2017-03-30 15-05-35

### PR DESCRIPTION
**Pull request by Achint08**
**Changes proposed in this pull request**
Fixes #313:-

 -  showing top hashtags on the home page

**Link to live demo:**
https://achint08.github.io/loklak_search/

**Screenshots**
screen shot 2017-06-22 at 1 00 18 am

@hemantjadon @kavithaenair @vibhcool @rishiraj824  Please review. :)

### CODE REVIEW

File: src/app/feed/feed-not-found/feed-not-found.component.html [hunk 1]
```diff
@@ -11,5 +11,7 @@
 			<li>Try different keywords, accounts or hashtags.</li>
 			<li>Try more general keywords, accounts or hashtags.</li>
 		</ul>
+		<p>If you encountered an error of the system, please submit a bug report in the
+			<a href="https://github.com/fossasia/loklak_search/issues/">loklak issue tracker here</a>.</p>
```
Comment (hemantjadon) [inline src/app/feed/feed-not-found/feed-not-found.component.html]: Not related To hashtags I guess?

File: src/app/home/home.component.ts [hunk 2]
```diff
@@ -61,6 +65,15 @@ export class HomeComponent implements OnInit, OnDestroy {
 		);
 	}
 
+	private getTopHashtags() {
+		this.store.dispatch(new queryAction.RelocationAfterQueryResetAction());
```
Comment (hemantjadon) [inline src/app/home/home.component.ts]: It is already reset No need of this

File: src/app/feed/feed-not-found/feed-not-found.component.html (hunk 1 above)
Comment (Achint08) [inline src/app/feed/feed-not-found/feed-not-found.component.html]: This is an another issue. #308

File: src/app/home/home.component.ts (hunk 2 above)
Comment (Achint08) [inline src/app/home/home.component.ts]: After user switches from feeds page to home page, it changes to set.

**Comment by Achint08** [comment]
@hemantjadon Removed the text `wait` and `Top Hashtags`. Lookout now.

File: src/app/feed/feed-not-found/feed-not-found.component.html (hunk 1 above)
Comment (hemantjadon) [inline src/app/feed/feed-not-found/feed-not-found.component.html]: Make separate PR's Don't make changes related to multiple issues at once, Please make a separate PR for this.

File: src/app/home/home.component.scss [hunk 3]
```diff
@@ -96,6 +96,18 @@ div.outer-wrapper {
...
+		padding-top:20px;
+
+		h4.heading{
+			text-align: center;
+		}
+
+		span.waiting-text{
+			padding-left:30px;
+		}
+	}
```
Comment (kavithaenair) [inline src/app/home/home.component.scss]: Identation problem.

**Comment by Achint08** [comment]
Done! checkout now.

File: src/app/home/home.component.html [hunk 4]
```diff
@@ -24,6 +24,16 @@
 			<span class="highlight">from:user</span> shows messages of a user;&nbsp;
 			<span class="highlight">followers:user</span> find followers of a user.
 		</div>
+
+		<div class="top-hashtags">
+			<div *ngIf="(apiResponseHashtags$ | async).length !== 0">
+					<span *ngFor ="let item of (apiResponseHashtags$ | async); let i = index">
```
Comment (hemantjadon) [inline src/app/home/home.component.html]: Use slice instead of iterating over the array.

**Comment by singhpratyush** [comment]
What do you guys think of this scheme? The background colour in HashTag is the primary colour of the app (`#d10042`). I think that font weight can be increased and there can be some elevation with secondary colour (`#f9f2f4`) on hover.

screenshot from 2017-06-23 12 05 52

I'm not asking to make the changes, just asking for views.

**Comment by Achint08** [comment]
@singhpratyush This would look good but mario want it in a simple text. Look at how google does that. :smile:

**Comment by Achint08** [comment]
@hemantjadon Checkout now. 
https://achint08.github.io/loklak_search/

**Comment by Achint08** [comment]
@kavithaenair Refresh the page once.

**Comment by singhpratyush** [comment]
@Achint08: The request made to get aggregated hashtags is this - 
http://api.loklak.org/api/search.json?q=since:day&callback=__ng_jsonp__.__req0.finished&minified=true&source=all&maximumRecords=20&timezoneOffset=-330&startRecord=1&fields=created_at,screen_name,mentions,hashtags&limit=10.

I think that this can be made better - 
* `source=all`
  Better to go with `source=cache` as querying `backend` or `twitter` will have no effect on aggregations, and the results will be generated faster as there will be no scraping taking place.
* `maximumRecords=20`
  You don't need them at all, you just need the aggregations - `maximumRecords=0`.
* `fields=created_at,screen_name,mentions,hashtags`
  Just `hashtags` will do. Running aggregation on many things will be expensive. `fields=hashtags`.
* `limit=10`
  If you are displaying just 5, it can be `limit=5`.

Please let me know about your view on this.

**Comment by vibhcool** [comment]
hey guys, I need just a hint where is api embedded in loklak search?

**Comment by Achint08** [comment]
@vibhcool look for src/app/services .

**Comment by Achint08** [comment]
@singhpratyush Thanks for this. 😄 
 I think as hemant is working on configurable service right now so @hemantjadon can look into this.

**Comment by hemantjadon** [comment]
@Achint08 Yes I will work on the configurable service.
One final thing we can do in this PR before merging is that Fixing the height of the hashtag box as right now the content is jumping here and there, Also do we really really require the text `Top hashtags in last 24 hours`, I mean isn't it feel obivious that there are latest hashtags.

**Comment by Achint08** [comment]
@hemantjadon Done! 👍

**Comment by Achint08** [comment]
@hemantjadon Check out please.
https://achint08.github.io/loklak_search/

**Comment by singhpratyush** [comment]
@Achint08 @hemantjadon: Could you please point me to an issue regarding the configurable service?

**Comment by hemantjadon** [comment]
@singhpratyush The issue is #425 and the corresponding PR which I have made is #430 Please check.

**Comment by Achint08** [comment]
@hemantjadon Are there some more changes that need to be introduced?

**Comment by mariobehling** [comment]
Ok, merging it as @hemantjadon is not responding and two approvals are there.
